src/resource_clients/request_queue.ts

```diff
@@ -39,14 +42,17 @@ export class RequestQueueClient extends ResourceClient {
         });
 
         this.clientKey = userOptions.clientKey;
-        this.timeoutMillis = userOptions.timeoutSecs ? userOptions.timeoutSecs * 1e3 : undefined;
+        this.timeoutMillis = userOptions.timeoutSecs;
```

Comment:
This looks confusing (millis / secs). Perhaps you want to rename the class field? (I see below you're using it as seconds).